tests/thread/enc/errno_tests.cpp

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#ifdef _PTHREAD_ENC_
+#include "thread.h"
+#endif
+
+#include <openenclave/enclave.h>
+#include <openenclave/internal/print.h>
+#include <openenclave/internal/thread.h>
+#include <openenclave/internal/types.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <algorithm>
+#include "errno.h"
+#include "thread_t.h"
+
+void* enc_malloc(size_t size, int* err)
+{
+    void* ret = malloc(size);
+    *err = errno;
```

Comment:
Note, standard C does not require that malloc set errno:
https://en.cppreference.com/w/c/memory/malloc

POSIX and Linux require that malloc set errno:
https://pubs.opengroup.org/onlinepubs/009695399/functions/malloc.html

Reply:
The following code is from "openenclave\enclave\core\malloc.c",  Does the malloc implementation in OE set errno?
void* oe_malloc(size_t size)
{
    void* p = MALLOC(size);

    if (!p && size)
    {
        errno = ENOMEM;

        if (_failure_callback)
            _failure_callback(__FILE__, __LINE__, __FUNCTION__, size);
    }

    return p;
}

Reply:
From what I understand, there is an inconsistency:

`oe_malloc` sets errno: https://github.com/openenclave/openenclave/blob/6bed4b5e0927746ca2e9f0c0c8b154e5dd66a204/enclave/core/malloc.c#L93-L106

`oe_debug_malloc` (which is used in debug builds) does not seem to set errno:
https://github.com/openenclave/openenclave/blob/6bed4b5e0927746ca2e9f0c0c8b154e5dd66a204/enclave/core/debugmalloc.c#L310-L327

Reply:
Let me know if you are running into failures due to this. We can provide a fix.